File: android/src/main/kotlin/com/gstory/flutter_unionad/FlutterUnionadViewPlugin.kt
package com.gstory.flutter_unionad

import android.app.Activity
import com.gstory.flutter_unionad.bannerad.BannerAdViewFactory
import com.gstory.flutter_unionad.drawfeedad.DrawFeedAdViewFactory
import com.gstory.flutter_unionad.nativead.NativeAdViewFactory
import com.gstory.flutter_unionad.splashad.SplashAdViewFactory
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * @Description:
 * @Author: gstory0404@gmail
 * @CreateDate: 2020/8/19 11:38
 */
object FlutterUnionadViewPlugin {
    fun registerWith(binding: FlutterPlugin.FlutterPluginBinding,activity :Activity) {
        //注册开屏广告
        binding.platformViewRegistry.registerViewFactory(FlutterunionadViewConfig.splashAdView, SplashAdViewFactory(binding.binaryMessenger,activity))
        //注册banner广告
        binding.platformViewRegistry.registerViewFactory(FlutterunionadViewConfig.bannerAdView, BannerAdViewFactory(binding.binaryMessenger,activity))
        //注册个性化模板信息流广告
        binding.platformViewRegistry.registerViewFactory(FlutterunionadViewConfig.nativeAdView, NativeAdViewFactory(binding.binaryMessenger,activity))
        //注册个性化模板draw视频广告
        binding.platformViewRegistry.registerViewFactory(FlutterunionadViewConfig.drawFeedAdView, DrawFeedAdViewFactory(binding.binaryMessenger,activity))
    }
}

File: ios/Classes/FlutterUnionadViewPlugin.swift
//
//  FlutterUnionadViewPlugin.swift
//  flutter_unionad
//
//  Created by gstory0404@gmail on 2020/9/4.
//

import Foundation
import Flutter

class FlutterUnionadViewPlugin{
    static func register(viewRegistrar : FlutterPluginRegistrar){
        print("初始化view")
        print(viewRegistrar.messenger())
       //banner广告
        let bannerAdViewFactory = BannerAdViewFactory(messenger: viewRegistrar.messenger())
        viewRegistrar.register(bannerAdViewFactory, withId: "com.gstory.flutter_unionad/BannerAdView")
        //spalsh广告
        let splashAdViewFactory = SplashAdViewFactory(messenger: viewRegistrar.messenger())
        viewRegistrar.register(splashAdViewFactory, withId: "com.gstory.flutter_unionad/SplashAdView")
        //draw广告
        let drawfeedAdViewFactory = DrawFeedAdViewFactory(messenger: viewRegistrar.messenger())
        viewRegistrar.register(drawfeedAdViewFactory, withId: "com.gstory.flutter_unionad/DrawFeedAdView")
        //信息流广告
        let nativeAdviewFactory = NativeAdViewFactory(messenger: viewRegistrar.messenger())
        viewRegistrar.register(nativeAdviewFactory, withId: "com.gstory.flutter_unionad/NativeAdView")
    }
}

